src/pepr/operator/controllers/keycloak/client-sync.ts

```diff
@@ -86,9 +92,9 @@ async function syncClient(
       metadata: {
         namespace: pkg.metadata!.namespace,
         // Use the CR secret name if provided, otherwise use the client name
-        name: clientReq.secretName || name,
+        name: secretName || name,
       },
-      stringData: clientToStringmap(client),
+      stringData: generateSecretData(client, secretTemplate),
```

Comment:
This currently acts as a patch, adding new keys to the secret but not deleting old ones if you edit a `secretTemplate` in your package CR. In some testing it seems like `data` (instead of `stringData`) operates more as expected, updating the secret contents and deleting old keys. Seems like we should switch to that?